search-service/src/main/kotlin/com/egm/stellio/search/web/EntityAccessControlHandler.kt

```diff
@@ -245,9 +248,15 @@ class EntityAccessControlHandler(
             val entityUri = entityId.toUri()
             authorizationService.userIsAdminOfEntity(entityUri, sub).bind()
 
-            val body = requestBody.awaitFirst()
-            val expandedPayload = expandJsonLdFragment(AUTH_TERM_SAP, body, COMPOUND_AUTHZ_CONTEXT)
+            val body = requestBody.awaitFirst().deserializeAsMap()
+            val contexts = checkAndGetContext(httpHeaders, body).addAuthzContextIfNeeded()
+            val rawPayload = mapOf(AUTH_TERM_SAP to JsonLdUtils.removeContextFromInput(body))
+            val expandedPayload = expandJsonLdFragment(rawPayload, contexts)
             val ngsiLdAttributes = parseToNgsiLdAttributes(expandedPayload)
+            if (ngsiLdAttributes[0].name != AUTH_PROP_SAP)
+                ResourceNotFoundException(
```

Comment:
I'm not really sure about `ResourceNotFoundException` A `BadRequestDataException` would be more appropriate ?

Reply:
indeed